packages/passport/sdk/src/zkEvm/types.ts

```diff
@@ -111,3 +111,68 @@ export type AccountsChangedEvent = Array<string>;
 export interface ProviderEventMap extends Record<string, any> {
   [ProviderEvent.ACCOUNTS_CHANGED]: [AccountsChangedEvent];
 }
+
+export interface EIP1193Provider {
```

Comment:
@Jon-Alonso I was looking around but we only have the `Provider` type and it has a number of variations that don't fit the EIP-1193 spec:

- `isPassport` is passport provider specific.
- `request()` arguments params value is defined as ` params?: Array<any>;` but should be defined as
``` readonly params?: readonly unknown[] | object;``` as per EIP-1193
- `send` and `sendAsync` are depreciated methods that are not part of the EIP-1193 interface.

As I'm adding support for the EIP-6963 spec and wanted to use a correct EIP-1193 interface.
One option here could be to rename `Provider` to `PassportProvider` and tweak the `ZkEvmProvider` to use this instead?

```
export class ZkEvmProvider implements EIP1193Provider, PassportProvider {
```

Reply:
@Jon-Alonso I outlined above why I didn't use the `Provider` type here because I was being explicit about exactly what EIP-1193 interface is as [defined here](https://eips.ethereum.org/EIPS/eip-1193).. I suggested a compromise to refactor the Provider to PassportProvider and include the interface. 
I understand the added stuff in Provider is harmless but it's not an EIP-1193 interface only which was my issue using that as EIP-6963 interface calls for that as well. I agree duplication is not desired hence my suggestion to refactor the type and be specific about both so it's clear.

But.. I've removed the interfaces I've added as you requested.

To be honest this simple PR has taken way to much time and energy +3 weeks. If you are happy with my changes I'll merge this stuff in, otherwise I'm going to leave this with you guys. Trying to merge this stuff is like pulling teeth.